### ISSUE TITLE
Add new Lighthouse VC API endpoint for updating Graffiti

### ISSUE THREAD

**Issue by jimmygchen** [sigp/lighthouse#4386]
## Description

Implement a new PUT endpoint in the Validator Client (VC) to support updating validator Graffiti in the Siren UI.

This is a bit tricky because there are currently multiple ways users can set the Graffiti:
1. [Using the "--graffiti-file" flag on the validator client](https://lighthouse-book.sigmaprime.io/graffiti.html#1-using-the---graffiti-file-flag-on-the-validator-client)
2. [Setting the graffiti in the validator_definitions.yml](https://lighthouse-book.sigmaprime.io/graffiti.html#2-setting-the-graffiti-in-the-validator_definitionsyml)
3. [Using the "--graffiti" flag on the validator client](https://lighthouse-book.sigmaprime.io/graffiti.html#3-using-the---graffiti-flag-on-the-validator-client)
4. [Using the "--graffiti" flag on the beacon node](https://lighthouse-book.sigmaprime.io/graffiti.html#4-using-the---graffiti-flag-on-the-beacon-node)

If the endpoint only updates the graffiti in-memory, it would be only temporary and the CLI/file override it when the node gets restated. For this to be persistent, we should probably write the changes to a file, as well as updating the value in-memory.

@michaelsproul suggested that we should write this to the [validator definitions](https://lighthouse-book.sigmaprime.io/graffiti.html#2-setting-the-graffiti-in-the-validator_definitionsyml), which is guaranteed to exist, and this is also consistent with the other HTTP API methods such as [`feerecipient`](https://github.com/sigp/lighthouse/blob/c547a11b0da48db6fdd03bca2c6ce2448bbcc3a9/validator_client/src/http_api/mod.rs#L734-L773). The only source that's higher priority is the `--graffiti-file` flag, so the endpoint could return an error if this value is set, to avoid overriding the changes.

To recap, the proposal: 
1. Add a new `PUT` Lighthouse API endpoint for updating graffiti, suggested path: `PUT lighthouse/validator/{pub key}/graffiti`
2. The endpoint writes the new graffiti to the `validator_definitions.yml` file for the specified public key
3. Update the validator graffiti in memory
4. If `--graffiti-file` is set, the endpoint should return an error indicating the change failed

**Comment by jimmygchen** [comment]
Completed in #4417 🎉
